### ISSUE TITLE
metrics-es-node.rb - 404 Resource Not Found

### ISSUE THREAD

**Issue by lfrancois** [sensu-plugins/sensu-plugins-elasticsearch#6]
Hi,

I think it used to work fine and I have an issue right now, I don't know if it's related to the upgrade of my  ES cluster to 1.7.0

```bash
root@esm1 ~ # which metrics-es-node.rb 
/usr/bin/metrics-es-node.rb
root@esm1 ~ # metrics-es-node.rb
Check failed to run: 404 Resource Not Found, ["/usr/lib/ruby/gems/1.9.1/gems/rest-client-1.8.0/lib/restclient/abstract_response.rb:74:in `return!'", "/usr/lib/ruby/gems/1.9.1/gems/rest-client-1.8.0/lib/restclient/request.rb:495:in `process_result'", "/usr/lib/ruby/gems/1.9.1/gems/rest-client-1.8.0/lib/restclient/request.rb:421:in `block in transmit'", "/usr/lib/ruby/1.9.1/net/http.rb:746:in `start'", "/usr/lib/ruby/gems/1.9.1/gems/rest-client-1.8.0/lib/restclient/request.rb:413:in `transmit'", "/usr/lib/ruby/gems/1.9.1/gems/rest-client-1.8.0/lib/restclient/request.rb:176:in `execute'", "/usr/lib/ruby/gems/1.9.1/gems/rest-client-1.8.0/lib/restclient/request.rb:41:in `execute'", "/usr/lib/ruby/gems/1.9.1/gems/rest-client-1.8.0/lib/restclient/resource.rb:51:in `get'", "/usr/lib/ruby/gems/1.9.1/gems/sensu-plugins-elasticsearch-0.1.1/bin/metrics-es-node.rb:61:in `run'", "/usr/lib/ruby/gems/1.9.1/gems/sensu-plugin-1.2.0/lib/sensu-plugin/cli.rb:56:in `block in <class:CLI>'"]
root@esm1 ~ # 
```
Any idea, what it could be?

**Comment by mattyjones** [comment]
@lfrancois  I know that Ruby <1.9.3 is not supported. With that being said you may have to do some tweaking but I would try it with the embedded Ruby. If that does not work the error message that you are getting refers to the URL that the check is trying to execute not being found.

I would start by making sure that url exists and if so try it with simple curl.

**Comment by lfrancois** [comment]
I use elasticsearch 1.7.1 and it seems there is not this nodes entry anymore under _cluster and it moved to _nodes. Is it normal?
```bash
root@esm1 ~ # curl -XGET 'http://localhost:9200/_cluster/nodes?pretty=true'
{
  "error" : "ElasticsearchIllegalArgumentException[No feature for name [nodes]]",
  "status" : 400
}
root@esm1 ~ # curl -XGET 'http://localhost:9200/_cluster/health?pretty=true'
{
  "cluster_name" : "esc1-prod",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 7,
  "number_of_data_nodes" : 3,
  "active_primary_shards" : 1244,
  "active_shards" : 2487,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0
}
root@esm1 ~ # 
```

It seems it moved here:
root@esm1 ~ # curl http://localhost:9200/_nodes/_local/stats

So I did that, and now it seems to work
```diff
root@esm1 [prod] [] /usr/lib/ruby/gems/1.9.1/gems/sensu-plugins-elasticsearch-0.1.1/bin # diff -u metrics-es-node.rb.orig metrics-es-node.rb
--- metrics-es-node.rb.orig	2015-08-10 04:51:28.263611769 +0000
+++ metrics-es-node.rb	2015-09-17 18:43:16.884514157 +0000
@@ -56,8 +56,8 @@
          default: 9200
 
   def run # rubocop:disable all
-    ln = RestClient::Resource.new "http://#{config[:host]}:#{config[:port]}/_cluster/nodes/_local", timeout: 30
-    stats = RestClient::Resource.new "http://#{config[:host]}:#{config[:port]}/_cluster/nodes/_local/stats", timeout: 30
+    ln = RestClient::Resource.new "http://#{config[:host]}:#{config[:port]}/_nodes/_local", timeout: 30
+    stats = RestClient::Resource.new "http://#{config[:host]}:#{config[:port]}/_nodes/_local/stats", timeout: 30
     ln = JSON.parse(ln.get)
     stats = JSON.parse(stats.get)
     timestamp = Time.now.to_i
root@esm1 [prod] [] /usr/lib/ruby/gems/1.9.1/gems/sensu-plugins-elasticsearch-0.1.1/bin # 
```

**Comment by mattk42** [comment]
It seems that this has already been fixed, but the latest release (0.2.0) doesn't include those changes? I pulled the latest code from master and it works as expected.

**Comment by eheydrick** [comment]
@lfrancois This plugin would need to be updated to work with ES 1.0+. I'd take a PR for the change you proposed above. Alternatively you could use metrics-es-node-graphite, which works with all versions of ES and collects way more metrics.

**Comment by mattk42** [comment]
Ah, I didn't see the difference between 'metrics-es-node-graphite' and 'metrics-es-node.rb'. That would be where my confusion was. Thanks!

**Comment by cstockton** [comment]
This affected me too, I opened https://github.com/sensu-plugins/sensu-plugins-elasticsearch/pull/19

**Comment by analytically** [comment]
Fixed in 0.3.1, can anyone confirm?

**Comment by eheydrick** [comment]
Confirmed working on 1.7. It did not work on 2.x so I added support and released 0.3.2.
